docs/rfcs/2023-09-29-merge-scenarios.md

```diff
@@ -0,0 +1,149 @@
+
+Probably the solution is to:
+- send probe to associated contact if it exists (implemented currently)
+- if associated contact doesn't exist send probe to group member (not implemented?)
+- send probe hashes to all matching contacts (implemented)
+- send probe hashes to all matching group members in other groups that don't have associated contact records (not implemented)
+- merge all contacts that confirmed (currently only the first confirming contact is merged)
+- merge all group members that confirmed (not implemented, merge profiles?)
+
+Check: if both group member and associated confirm probe, will they be properly merged?
+
+### Connecting via profile address
+
+There are two possible scenarios: contact request is made via member info (associated group member is already in UI scope); and contact request is made via any other means but contact address is known.
+
+**1. Contact request is made via button in group member info**
+
+(contact chat info doesn't have such button currently, as we're already in a chat with contact)
+
+Contact record can be associated directly to a group member record.
+
+It can be simplified to reusing "send direct message" logic here (though there should be no message if direct messages preference is off in group), or even hiding this button now that "send direct message" exists.
+
+To consider:
+
+Currently group member address is shown even if direct messages are prohibited in group. There're two reasons this preference is usually enabled in a group:
+- to prevent abuse (in public groups),
+- to prevent members from direct communication.
+
+Member address being shown regardless of this preference undermines the second use case.
+
+**2. Contact request is made via any other means**
+
+- On any contact request, search contact_profiles table for matching address (add index on contact_link).
+
+- If profile is not found:
+
+  - Send contact request as usual.
+
+- If profile is found:
+
+  - Search for contact by contact_profile_id.
+
+  - If contact is not found:
+
+    - Send contact request, associating it with existing profile.
+    - What if contact is accepted with a different profile? Update profile?
+
+  - If contact is found:
+
+    - Open existing contact.
+
+It's unlikely but possible (?) that contact deleted their contact address, didn't notify about profile change, then another user created link with the same address. If it is then used to make a contact request wrong record may be associated. Probability of that is very low, but maybe we shouldn't make such direct associations and instead always rely on probing?
+
+### Repeat group join via group link
+
+**If group still exists:**
+
+Open group.
+
+How to check for group existence? Currently we save group_link_id on host contact's connection. It may have been deleted by the time of repeated connection via group link. Probably we should also save group_link_id or even the full contact address on the group record itself.
+
+**If group doesn't exist:**
```

Comment:
If group doesn't exist and contact exists - just allow repeat contact request, and trigger probing.